libs/server/Storage/Session/ObjectStore/SortedSetOps.cs

```diff
@@ -692,6 +692,108 @@ public GarnetStatus SortedSetAdd<TObjectContext>(byte[] key, ref ObjectInput inp
         where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
         => RMWObjectStoreOperationWithOutput(key, ref input, ref objectStoreContext, ref output);
 
+        /// <summary>
+        /// ZRANGESTORE - Stores a range of sorted set elements into a destination key.
+        /// </summary>
+        /// <typeparam name="TObjectContext">The type of the object context.</typeparam>
+        /// <param name="distKey">The destination key where the range will be stored.</param>
+        /// <param name="sbKey">The source key from which the range will be taken.</param>
+        /// <param name="input">The input object containing range parameters.</param>
+        /// <param name="result">The result of the operation, indicating the number of elements stored.</param>
+        /// <param name="objectStoreContext">The context of the object store.</param>
+        /// <returns>Returns a GarnetStatus indicating the success or failure of the operation.</returns>
+        public unsafe GarnetStatus SortedSetRangeStore<TObjectContext>(ArgSlice distKey, ArgSlice sbKey, ref ObjectInput input, out int result, ref TObjectContext objectStoreContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            result = 0;
+
+            if (distKey.Length == 0 || sbKey.Length == 0)
+                return GarnetStatus.OK;
+
+            var createTransaction = false;
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(distKey, true, LockType.Exclusive);
+                txnManager.SaveKeyEntryToLock(sbKey, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+
+            // SetObject
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            try
+            {
+                SpanByteAndMemory rangeOutputMem = default;
+                var rangeOutput = new GarnetObjectStoreOutput() { spanByteAndMemory = rangeOutputMem };
+                var status = SortedSetRange(sbKey.ToArray(), ref input, ref rangeOutput, ref objectStoreLockableContext);
```

Comment:
I wonder if it would be worthwhile performance-wise to do a GET and do the range logic here instead of parsing the output. You could probably write a method in SortedSetObject that takes an ObjectInput and returns an IEnumerable<double, byte[]>, and use it in both implementations (ZRANGE & ZRANGESCORE). WDYT?

Reply:
Performance is of higher priority than code sharing, IMHO.

Reply:
@badrishc I agree. Off-topic, sorry to bring it up here. I have a question for you here https://github.com/microsoft/garnet/pull/674#issuecomment-2515441660

Reply:
You're correct, we shouldn't incur any performance cost on ZRANGE. It would still be nice to at least make some portion of it common, if possible, since it's a rather hefty chunk of code, even if just the parsing / validation of options.

Reply:
Let me think about it, currently, I am not sure how I can do that

Reply:
@TalZaccai I am not sure we can do this.

> You could probably write a method in SortedSetObject that takes an ObjectInput and returns an IEnumerable<double, byte[]>

We can return `SpanByteAndMemory` from the SortedSetObject class, right? Then how can we return anything like `IEnumerable<double, byte[]>`. Maybe I didn't understand the review command.

Reply:
Not sure I follow you... What I meant was that we could do something like:
```
-- begin transaction
srcSortedSetObj = GET(srcKey)
(items, scores) = srcSortedSetObj.GetRange(input) // new method
dstSortedSetObj = GET(dstKey)
dstSortedSetObj.Add((items, scores))
-- end transaction
```

Reply:
It's fine as-is, I was just wondering if we could squeeze a bit better perf here. No worries if you want to skip it for now. LMK.

Reply:
Ohh now only I understood your intent. Never thought we were allowed to write public methods in SortedSetObject (any Object types) and call it directly. If I have to write the new `GetRange(input)` method then most of the code with the current `SortedSetRange` method will be duplicated. I can't see a way to have a common code without allocating a new object or many dirty condition checks to check if we need to write the output to SpanByteAndMemory or a new object.

If you are fine with having lots of duplicate code then I can do it. Or if you have an idea to have a common code without affecting the performance of ZRANGE then let me know as well.

Reply:
I didn't mean that it would replace SortedSetRange, but that SortedSetRange would also use that helper method, and it could contain only part of the logic there (such as parsing) to avoid code duplication. I'll just let this in as-is and we can look into improving perf on this at a later stage.